monash-human-power/dashboard
# Debashishsahoo/anemometer data on dashboard

## Description

Displays anemometer data in a new card on the status page of the dashboard (wind direction, wind speed, and online status).

The Anemometer card design is similar to the existing Wireless Modules card on the status page. The data is retrieved and extracted by subscribing to the Wireless Module 5 topic using MQTT. The functionality was tested using Marcus Chow's mock wind data. Refer to the steps below for a description of the whole testing process. Manually publishing the online status to be true for the wireless module in Step 5 is crucial and is a new step added to get the mock wind data testing to actually work and display the data on the dashboard.

## Screenshots

![Anemometer Data on Dashboard](https://github.com/monash-human-power/dashboard/assets/69211573/f1b3ef7f-4a5a-4b8e-b690-25ab841a1ea1)

## Steps to Test

Open five different terminals and run each set of commands in its own terminal:

1.
- Navigate to `dashboard/server` 
- `yarn start` (assuming dependencies and libraries have been installed)
2. 
- Navigate to `dashboard/client` 
- `yarn start` (assuming dependencies and libraries have been installed)
3.
- Clone the DAS repository and navigate to `DAS/data-acquisition-system/DAS`
- `poetry env use "ENTER YOUR LOCAL PATH TO PYTHON"`
- `poetry install` 
- `poetry shell`
- `cd das`
- `python V3_fake_module.py`
4.
- `mosquitto_sub -t "/v3/wireless_module/5/data"` to separately see the mock wind data being posted from the fake script.
5.
- `mosquitto_pub -t "/v3/wireless_module/5/status" -m '{"online": true}'` to set the online status of the anemometer wireless module to true in order to display the data on the dashboard.
